make htmldocs-fast not working in development workspace

`make htmldocs-fast` fails for me with the following error:

```
tirzah[262]$ pwd
/mnt/nordic/zp/zephyr
tirzah[263]$ rm -rf doc/_build/
tirzah[264]$ make htmldocs-fast
make htmldocs KCONFIG_TURBO_MODE=1
make[1]: Entering directory '/mnt/nordic/zp/zephyr'
cmake -GNinja  -Bdoc/_build -Sdoc/ -DDOC_TAG=development \
	-DSPHINXOPTS=-q \
	-DKCONFIG_TURBO_MODE=1
-- Zephyr base: /mnt/nordic/zp/zephyr
-- West: /home/pab/.local/bin/west (west --version: "West version: v0.8.0")
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.5") found components: Interpreter 
-- Found Doxygen: /usr/local/bin/doxygen (found version "1.9.0 (8e87f6bb404065d2c8802a9c61756e3c7a729fb1)") found components: doxygen 
-- Found LATEX: /usr/bin/latex   
-- Zephyr version: 2.4.99 (/mnt/nordic/zp/zephyr)
CMake Error at /mnt/nordic/zp/zephyr/cmake/zephyr_module.cmake:107 (zephyr_string):
  Unknown CMake command "zephyr_string".
Call Stack (most recent call first):
  CMakeLists.txt:47 (include)
-- Configuring incomplete, errors occurred!
See also "/mnt/nordic/zp/zephyr/doc/_build/CMakeFiles/CMakeOutput.log".
make[1]: *** [Makefile:25: configure] Error 1
make[1]: Leaving directory '/mnt/nordic/zp/zephyr'
make: *** [Makefile:16: htmldocs-fast] Error 2
```

On slack @utzig fixed this with the following patch:
```
diff --git a/doc/CMakeLists.txt b/doc/CMakeLists.txt
index 4c34e20e67..6fb3274851 100644
--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -44,6 +44,7 @@ include(${ZEPHYR_BASE}/cmake/version.cmake)
 # Process modules
 set(KCONFIG_BINARY_DIR ${CMAKE_BINARY_DIR}/Kconfig)
 file(MAKE_DIRECTORY ${KCONFIG_BINARY_DIR})
+include(${ZEPHYR_BASE}/cmake/extensions.cmake)
 include(${ZEPHYR_BASE}/cmake/zephyr_module.cmake)
 # Note that this won't force fatal error if latexmk is not found.
```

Once that's done, I get this error:
```
tirzah[268]$ make htmldocs-fast
make htmldocs KCONFIG_TURBO_MODE=1
make[1]: Entering directory '/mnt/nordic/zp/zephyr'
cmake -GNinja  -Bdoc/_build -Sdoc/ -DDOC_TAG=development \
	-DSPHINXOPTS=-q \
	-DKCONFIG_TURBO_MODE=1
-- Zephyr base: /mnt/nordic/zp/zephyr
-- West: /home/pab/.local/bin/west (west --version: "West version: v0.8.0")
-- Zephyr version: 2.4.99 (/mnt/nordic/zp/zephyr)
-- Configuring done
-- Generating done
-- Build files have been written to: /mnt/nordic/zp/zephyr/doc/_build
cmake --build doc/_build -- htmldocs  # -v # VERBOSE=1
[0/5] Running gen_devicetree_rest.py /mnt/nordic/zp/zephyr/doc/_build/rst/doc/reference/devicetree
[1/5] Running gen_kconfig_rest.py /mnt/nordic/zp/zephyr/doc/_build/rst/doc/reference/kconfig/
FAILED: CMakeFiles/kconfig 
cd /mnt/nordic/zp/zephyr/doc && /usr/bin/cmake -E make_directory /mnt/nordic/zp/zephyr/doc/_build/rst/doc/reference/kconfig && /usr/bin/cmake -E env PYTHONPATH=/mnt/nordic/zp/zephyr/scripts/kconfig:.:/home/pab/Utils/lib/python:/usr/local/lib64/python3.8/site-packages:/usr/local/lib/python3.8/site-packages:/mnt/nordic/zp/zephyr/scripts/dts ZEPHYR_BASE=/mnt/nordic/zp/zephyr srctree=/mnt/nordic/zp/zephyr "BOARD_DIR=boards/*/*" "ARCH=*" ARCH_DIR=arch SOC_DIR=soc KCONFIG_BINARY_DIR=/mnt/nordic/zp/zephyr/doc/_build/Kconfig KCONFIG_WARN_UNDEF=y KCONFIG_TURBO_MODE=1 KCONFIG_DOC_MODE=1 /usr/bin/python3.8 scripts/gen_kconfig_rest.py /mnt/nordic/zp/zephyr/doc/_build/rst/doc/reference/kconfig/ --separate-all-index --keep-module-paths --modules Architecture,arch,/mnt/nordic/zp/zephyr/arch Driver,drivers,/mnt/nordic/zp/zephyr/drivers Kernel,kernel,/mnt/nordic/zp/zephyr/kernel Library,lib,/mnt/nordic/zp/zephyr/lib Subsystem,subsys,/mnt/nordic/zp/zephyr/subsys "External Module,modules,/mnt/nordic/zp/zephyr/modules"
scripts/gen_kconfig_rest.py: doc/_build/Kconfig/Kconfig.modules:2: Could not open '/mnt/nordic/zp/zephyr/' (in 'osource "$(ZEPHYR_HAL_NORDIC_KCONFIG)"') (EISDIR: Is a directory)
[3/5] Copying files to /mnt/nordic/zp/zephyr/doc/_build/rst
ninja: build stopped: subcommand failed.
make[1]: *** [Makefile:19: htmldocs] Error 1
make[1]: Leaving directory '/mnt/nordic/zp/zephyr'
make: *** [Makefile:16: htmldocs-fast] Error 2
```

which looks a lot like what #31446 fixed for another script.  I don't know where to fix that.

--- a/doc/CMakeLists.txt
+++ b/doc/CMakeLists.txt
@@ -43,7 +43,9 @@ endif()
 include(${ZEPHYR_BASE}/cmake/version.cmake)
 # Process modules
 set(KCONFIG_BINARY_DIR ${CMAKE_BINARY_DIR}/Kconfig)
+list(INSERT MODULE_EXT_ROOT 0 ${ZEPHYR_BASE})
 file(MAKE_DIRECTORY ${KCONFIG_BINARY_DIR})
+include(${ZEPHYR_BASE}/cmake/extensions.cmake)
 include(${ZEPHYR_BASE}/cmake/zephyr_module.cmake)
 
 # Note that this won't force fatal error if latexmk is not found.
@@ -224,6 +226,21 @@ file(WRITE ${KCONFIG_BINARY_DIR}/Kconfig.soc.arch
      "osource \"${ZEPHYR_BASE}/soc/$(ARCH)/*/Kconfig\"\n"
 )
 
+foreach(module_name ${ZEPHYR_MODULE_NAMES})
+  zephyr_string(SANITIZE TOUPPER MODULE_NAME_UPPER ${module_name})
+  list(APPEND
+       ZEPHYR_KCONFIG_MODULES
+       "ZEPHYR_${MODULE_NAME_UPPER}_MODULE_DIR=${ZEPHYR_${MODULE_NAME_UPPER}_MODULE_DIR}"
+  )
+
+  if(ZEPHYR_${MODULE_NAME_UPPER}_KCONFIG)
+    list(APPEND
+         ZEPHYR_KCONFIG_MODULES
+         "ZEPHYR_${MODULE_NAME_UPPER}_KCONFIG=${ZEPHYR_${MODULE_NAME_UPPER}_KCONFIG}"
+  )
+  endif()
+endforeach()
+
 add_custom_target(
   kconfig
   COMMAND ${CMAKE_COMMAND} -E make_directory ${RST_OUT}/doc/reference/kconfig
@@ -239,6 +256,7 @@ add_custom_target(
   KCONFIG_WARN_UNDEF=y
   KCONFIG_TURBO_MODE=${KCONFIG_TURBO_MODE}
   KCONFIG_DOC_MODE=1
+  ${ZEPHYR_KCONFIG_MODULES}
     ${PYTHON_EXECUTABLE} scripts/gen_kconfig_rest.py ${RST_OUT}/doc/reference/kconfig/
       --separate-all-index
       --keep-module-paths

